src/python/pants/bin/daemon_pants_runner.py

```diff
@@ -24,11 +24,12 @@
 from pants.util.socket import teardown_socket
 
 
-class NoopExiter(Exiter):
+class PantsRunFailCheckerExiter(Exiter):
+  """Passed to pants runs triggered from this class, will raise an exception if the pants run failed."""
+
   def exit(self, result, *args, **kwargs):
     if result != 0:
-      # TODO: https://github.com/pantsbuild/pants/issues/7652
-      raise _GracefulTerminationException(result)
+      raise _PantsRunFinishedWithFailureException(result)
```

Comment:
should we just call it `_PantsRunFinishedWithException`

Reply:
I think this would me misleading (because a pants run can finish with an error but not necessarily throw an exception). I'd rather call it `_PantsRunFinishedWithFailure`.